lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -2281,11 +2281,11 @@ public function getDefaultValueDeclarationSQL($field)
             if (isset($field['type'])) {
                 if (in_array((string) $field['type'], array("Integer", "BigInt", "SmallInt"))) {
                     $default = " DEFAULT ".$field['default'];
-                } elseif (in_array((string) $field['type'], array('DateTime', 'DateTimeTz')) && $field['default'] == $this->getCurrentTimestampSQL()) {
+                } elseif (in_array((string) $field['type'], array('DateTime', 'DateTimeTz', 'DateTimeImmutable', 'DateTimeTzImmutable')) && $field['default'] == $this->getCurrentTimestampSQL()) {
                     $default = " DEFAULT ".$this->getCurrentTimestampSQL();
-                } elseif ((string) $field['type'] == 'Time' && $field['default'] == $this->getCurrentTimeSQL()) {
+                } elseif (in_array((string) $field['type'], array('Time', 'TimeImmutable')) && $field['default'] == $this->getCurrentTimeSQL()) {
```

Comment:
Please use short-array syntax here

Reply:
Please use the third parameter in `in_array()` and is it possible to use strict comparison on the `$field['default']` instead?